tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -0,0 +1,143 @@
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from typing import Optional
+from rptest.clients.rpk import RpkTool, TopicSpec
+from rptest.services.spark_service import SparkService
+from rptest.services.trino_service import TrinoService
+from rptest.tests.datalake.iceberg_rest_catalog import IcebergRESTCatalogTest
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from ducktape.utils.util import wait_until
+
+
+class DatalakeServicesBase(IcebergRESTCatalogTest):
+    """All inclusive base class for implementing datalake tests. Includes the
+    boiler plate to enable datalake and bring up related services.
+    """
+    def __init__(self,
+                 test_ctx,
+                 filesystem_catalog_mode=True,
+                 *args,
+                 **kwargs):
+        super(DatalakeServicesBase,
+              self).__init__(test_ctx,
+                             extra_rp_conf={
+                                 "iceberg_enabled": "true",
+                                 "iceberg_catalog_commit_interval_ms": 5000
+                             },
+                             *args,
+                             **kwargs)
+        self.test_ctx = test_ctx
+        self.filesystem_catalog_mode = filesystem_catalog_mode
+        self.spark: Optional[SparkService] = None
+        self.trino: Optional[TrinoService] = None
+
+    def setUp(self):
+        self.catalog_service.set_filesystem_wrapper_mode(
+            self.filesystem_catalog_mode)
+        super().setUp()
+        si = self.redpanda.si_settings
+        self.spark = SparkService(
+            self.test_ctx,
+            iceberg_catalog_rest_uri=str(self.catalog_service.catalog_url),
+            cloud_storage_access_key=str(si.cloud_storage_access_key),
+            cloud_storage_secret_key=str(si.cloud_storage_secret_key),
+            cloud_storage_region=si.cloud_storage_region,
+            cloud_storage_api_endpoint=str(si.endpoint_url))
+        self.spark.start()
+        self.trino = TrinoService(
+            self.test_ctx,
+            iceberg_catalog_rest_uri=str(self.catalog_service.catalog_url),
+            cloud_storage_access_key=str(si.cloud_storage_access_key),
+            cloud_storage_secret_key=str(si.cloud_storage_secret_key),
+            cloud_storage_region=si.cloud_storage_region,
+            cloud_storage_api_endpoint=str(si.endpoint_url))
+        self.trino.start()
+
+    def tearDown(self):
+        if self.spark:
+            self.spark.stop()
+        if self.trino:
+            self.trino.stop()
+        return super().tearDown()
+
+    def create_iceberg_enabled_topic(self,
+                                     name,
+                                     partitions=1,
+                                     replicas=1,
```

Comment:
nit: maybe rather than a method to create the topic, could we have a method to add iceberg properties to a config, so that tests can extend the configs with other topic properties? It's already trivial to create topics, but it's less trivial to construct configs. I'm thinking that we may want to test other topic features together with Iceberg